pkg/clients/rds/rds.go

```diff
@@ -435,14 +444,11 @@ func LateInitialize(in *v1beta1.RDSInstanceParameters, db *rds.DBInstance) { //
 }
 
 // IsUpToDate checks whether there is a change in any of the modifiable fields.
-func IsUpToDate(p v1beta1.RDSInstanceParameters, db rds.DBInstance) (bool, error) {
-	// TODO(muvaf): ApplyImmediately and other configurations that exist in
-	//  <Modify/Create/Delete>DBInstanceInput objects but not in DBInstance
-	//  object are not late-inited. So, this func always returns true when
-	//  those configurations are changed by the user.
-
-	// TODO(muvaf): If a secret is provided for password, this logic should check
-	// whether it's changed by comparing it to the password in the published secret.
+func IsUpToDate(ctx context.Context, kube *client.Client, p v1beta1.RDSInstanceParameters, db rds.DBInstance, outCreds *v1alpha1.SecretReference) (bool, error) {
```

Comment:
```suggestion
func IsUpToDate(ctx context.Context, kube *client.Client, p v1beta1.RDSInstance, db rds.DBInstance) (bool, error) {
```

What do you think about supplying the whole `RDSInstance` and get the necessart secret refs in the function?

Reply:
```suggestion
func IsUpToDate(ctx context.Context, kube client.Client, p v1beta1.RDSInstanceParameters, db rds.DBInstance, outCreds *v1alpha1.SecretReference) (bool, error) {
```

Since `client.Client` is an interface, we don't need to send it via pointer.